File: day6.py
f = open('inputs/day6.txt', 'r')

input = f.read().strip()
rawBlocks = input.split("\t")
blocks = [ int(x) for x in rawBlocks ]


def balanceBlocks(blocks, index):
    #print(blocks)
    blocksToRedistribute = blocks[index]
    blocks[index] = 0
    curIndex = index
    while(blocksToRedistribute > 0):
        curIndex += 1
        if(curIndex == len(blocks)): curIndex = 0
        blocks[curIndex] += 1
        blocksToRedistribute -= 1
    return blocks

count = 0
banks = []
curBank = ''.join(str(x) for x in blocks)
while (not curBank in banks):
    banks.append(curBank)
    count += 1
    blocks = balanceBlocks(blocks, blocks.index(max(blocks)))
    curBank = ''.join(str(x) + ',' for x in blocks)
print("PART1")
print(count)

count = 0
banksTwo = []
while (not curBank in banksTwo):
    banksTwo.append(curBank)
    count += 1
    blocks = balanceBlocks(blocks, blocks.index(max(blocks)))
    curBank = ''.join(str(x) + ',' for x in blocks)
print("PART2")
print(count)

File: day6.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace day6dotnet
{
    class Day6
    {
        static void Main(string[] args)
        {
            string input = System.IO.File.ReadAllText("day6.txt");
            Console.WriteLine(input);

            List<int> blocks = input.Split('\t').Select(Int32.Parse).ToList();

            int count = 0;
            List<string> oldPatterns = new List<string>();
            string curPattern = String.Empty;
            while (!oldPatterns.Contains(curPattern))
            {
                if(curPattern != String.Empty) oldPatterns.Add(curPattern));
                Console.WriteLine(curPattern);
                count++;
                // Get maximum item
                int maxIndex = blocks.IndexOf(blocks.Max());
                int maxValue = blocks.Max();
                // Redistribute
                int curI = maxIndex < blocks.Count() ? maxIndex + 1 : 0;
                int iValue = maxValue % blocks.Count;
                //blocks.Select((x, i) => i == maxIndex ? iValue : i + increment);
                while (maxValue > 0)
                {
                    curI = curI < blocks.Count() - 1 ? curI + 1 : 0;
                    blocks[curI] += 1;
                    maxValue -= 1;
                }
                curPattern = string.Join(",", blocks.ToArray());
                Console.WriteLine(curPattern);

            }
            Console.WriteLine(String.Format("[PART1] : Count {0}", count));
            Console.Read();
        }
    }
}
